patches/server/0807-Add-PlayerShieldDisableEvent.patch

```diff
@@ -0,0 +1,75 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Cryptite <cryptite@gmail.com>
+Date: Mon, 20 Sep 2021 13:27:50 -0500
+Subject: [PATCH] Add PlayerShieldDisableEvent
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/Mob.java b/src/main/java/net/minecraft/world/entity/Mob.java
+index 8a864238e154e2131834d013652746b7e7a78c97..91afdf9b0bfd44e2dbfd1c36cbd35b50a6fbb407 100644
+--- a/src/main/java/net/minecraft/world/entity/Mob.java
++++ b/src/main/java/net/minecraft/world/entity/Mob.java
+@@ -1588,9 +1588,16 @@ public abstract class Mob extends LivingEntity {
+             float f = 0.25F + (float) EnchantmentHelper.getBlockEfficiency(this) * 0.05F;
+ 
+             if (this.random.nextFloat() < f) {
++                // Paper start
++                io.papermc.paper.event.player.PlayerShieldDisableEvent event = new io.papermc.paper.event.player.PlayerShieldDisableEvent((org.bukkit.entity.Player) player.getBukkitEntity(), getBukkitEntity());
++                event.callEvent();
++
++                if (!event.isCancelled()) {
+                 player.getCooldowns().addCooldown(Items.SHIELD, 100);
+                 this.level.broadcastEntityEvent(player, (byte) 30);
+             }
++                // Paper end
++            }
+         }
+ 
+     }
+diff --git a/src/main/java/net/minecraft/world/entity/player/Player.java b/src/main/java/net/minecraft/world/entity/player/Player.java
+index cea92f1dc663bf0648b2bd877d86ca380a517bc9..450fe792114e40d6fbe7ba79ad4d419d36f3fe6b 100644
+--- a/src/main/java/net/minecraft/world/entity/player/Player.java
++++ b/src/main/java/net/minecraft/world/entity/player/Player.java
+@@ -936,7 +936,7 @@ public abstract class Player extends LivingEntity {
+     protected void blockUsingShield(LivingEntity attacker) {
+         super.blockUsingShield(attacker);
+         if (attacker.getMainHandItem().getItem() instanceof AxeItem) {
+-            this.disableShield(true);
++            this.disableShield(true, attacker); // Paper
+         }
+ 
+     }
+@@ -1403,6 +1403,11 @@ public abstract class Player extends LivingEntity {
+     }
+ 
+     public void disableShield(boolean sprinting) {
++        //Paper start
```

Comment:
This diff should be re-done following the concept I linked on the prior PR: See [here](https://github.com/PaperMC/Paper/blob/master/patches/api/0165-Support-cancellation-supression-of-EntityDismount-Ve.patch#L37) for an example on how to create an overloaded method using paper comments.

Basic concept is to start the `// Paper start` block like you already do and close it right after the new method declaration.